modules/end-user-guide/examples/snip_che-configuring-a-workspace-to-use-an-intellij-based-ide.adoc

```diff
@@ -0,0 +1,71 @@
+.Prerequisites
+
+* Define the IntelliJ-based IDE container image to use: __<IntelliJ-container-image>__. It can be:
+** A prebuilt IntelliJ-based IDEs image:
++
+`quay.io/che-incubator/che-idea:latest`:: IntelliJ IDEA Community Edition
++
+`quay.io/che-incubator/che-pycharm:latest`:: PyCharm Community Edition
+** A custom editor image, see xref:building-intellij-editor-images.adoc[].
+* Define the destination to publish the editor metadata.
++
+`__<URL_to_meta.yaml>__`:: HTTPS resource containing the `meta.yaml` file defining the IntelliJ-based IDE.
+
+
+.Procedure
+
+. Create a `+meta.yaml+` file with the following content, and publish it to `__<URL_to_meta.yaml>__`:
++
+[source,yaml,subs="+quotes,macros,attributes"]
+----
+apiVersion: v2
+publisher: <username>
+name: intellij-ide
+version: latest
+type: Che Editor
+displayName: IntelliJ-based IDE
+title: IntelliJ-based IDE as Editor for Eclipse Che
+description: IntelliJ-based IDE running using Projector
+icon: https://www.jetbrains.com/apple-touch-icon.png
+category: Editor
+repository: https://github.com/che-incubator/jetbrains-editor-images
+firstPublicationDate: "2021-04-10"
+spec:
+  endpoints:
+   -  name: "intellij"
+      public: true
+      targetPort: 8887
+      attributes:
+        protocol: http
+        type: ide
+        path: /projector-client/index.html?backgroundColor=434343&wss
+  containers:
+   - name: intellij-ide
+     image: "__<IntelliJ-container-image>__"
+     mountSources: true
+     volumes:
+         - mountPath: "/home/projector-user"
+           name: projector-user
+     ports:
+         - exposedPort: 8887
+     memoryLimit: "4096M"
+----
+
+. Add the following component to the workspace devfile:
```

Comment:
Is it sufficient? On workspaces.openshift.com the workspace fails with following error:

```
Danger alert:Workspace che-docs-ivmy9 failed to start
Unrecoverable event occurred: 'FailedCreate', 'Error creating: pods "workspaceeuj9zsz1y9ukwzxf.che-docs-cff7466bf-67hbw" is forbidden: exceeded quota: for-ffloreth-compute, requested: limits.memory=9267871744, used: limits.memory=0, limited: limits.memory=7Gi', 'workspaceeuj9zsz1y9ukwzxf.che-docs-cff7466bf'
```

Reply:
@vzhukovs please reconsider the memory limits taken into account that this functionality is expected to work on workspaces.openshift.com - https://github.com/eclipse/che/issues/20313

Reply:
> @vzhukovs please reconsider the memory limits taken into account that this functionality is expected to work on workspaces.openshift.com - [eclipse/che#20313](https://github.com/eclipse/che/issues/20313)

The latest my attempt, shown that it worked on workspaces.openshift.com. I'll re-check it later today again.

Reply:
Opening https://workspaces.openshift.com#https://github.com/themr0c/che-docs/tree/test-intellij

I had following:

```
Unknown workspace error.
```

And in the logs:

```
Failed to run the workspace: "Your session has expired. Please <a href='javascript:location.reload();' target='_top'>login</a> to Che again to get access to your OpenShift account"
```

Reply:
Then again:  

```
All plugin metadata has been successfully processed

Failed to run the workspace: "Unrecoverable event occurred: 'FailedCreate', 'Error creating: pods "workspacecrmtmgizopsetocl.che-docs-66bc9ff57d-7pvkw" is forbidden: exceeded quota: for-ffloreth-compute, requested: limits.memory=9402089472, used: limits.memory=0, limited: limits.memory=7Gi', 'workspacecrmtmgizopsetocl.che-docs-66bc9ff57d'"
```

I am using the latest stable meta for that branch.

```
  - type: cheEditor
    reference: https://github.com/che-incubator/jetbrains-editor-images/raw/meta/che-idea/latest.meta.yaml
```

In the devfile, we are setting limits only for the che-docs container (512M), all  the rest is using default values..

Reply:
Trying with the NodeJS Express stack. The workspace started correctly with the default editor. It doesn't start with IntelliJ: session expired multiple times.

Reply:
> - type: cheEditor
>     reference: https://github.com/che-incubator/jetbrains-editor-images/raw/meta/che-idea/latest.meta.yaml

Try to use links to the meta yaml from the following devfiles:

- https://github.com/che-incubator/jetbrains-editor-images/blob/main/devfiles/latest/che-idea/latest.yaml
- https://github.com/che-incubator/jetbrains-editor-images/blob/main/devfiles/latest/che-pycharm/latest.yaml

With this devfiles workspaces successfully start on workspaces.openshift.com

Because, the link provided in your comment performs the redirect:
>curl https://github.com/che-incubator/jetbrains-editor-images/raw/meta/che-idea/latest.meta.yaml
<html><body>You are being <a href="https://raw.githubusercontent.com/che-incubator/jetbrains-editor-images/meta/che-idea/latest.meta.yaml">redirected</a>.</body></html>

Reply:
Now I understand why  raw.githubusercontent.com is preferable!

Let's use the  raw.githubusercontent.com URLs, for consistencye with the rest of the docs (instead  of  github.com...raw... and github.com...blob)